exts/cesium.omniverse/cesium/omniverse/ui/credits_parser.py

```diff
@@ -2,60 +2,69 @@
 import omni.kit.pipapi
 import omni.ui as ui
 import webbrowser
-from typing import Optional, List, Tuple
+from typing import Optional, List, Tuple, NamedTuple
 from .uri_image import CesiumUriImage
 from .image_button import CesiumImageButton
 
 _num_retries = 0
 
 
+class ParsedCredit(NamedTuple):
+    text: Optional[str] = None
+    image_uri: Optional[str] = None
+    link: Optional[str] = None
+
+
 class CesiumCreditsParser:
     """Takes in a credits array and outputs the elements necessary to show the credits.
     Should be embedded in a VStack or HStack."""
 
-    def _parse_element(self, element, link: Optional[str] = None):
+    def _parse_element(self, element, results: List[ParsedCredit], link: Optional[str] = None):
```

Comment:
I'm really not a huge fan of passing around an array (`results`) and mutating it. This kind of mutation of state almost always leads to bugs and mistakes as time goes on. Can you return a `ParsedCredit` instead?

Reply:
> The function could return List[ParsedCredit] instead

I ended up doing this.